pkg/server/server_test.go

```diff
@@ -60,13 +56,20 @@ func server() {
 	}
 
 	api.RegisterRegistryServer(s, NewRegistryServer(store))
-	if err := s.Serve(lis); err != nil {
-		logrus.Fatalf("failed to serve: %v", err)
-	}
+	return s
 }
 
 func TestMain(m *testing.M) {
-	go server()
+	lis, err := net.Listen("tcp", port)
+	if err != nil {
+		logrus.Fatalf("failed to listen: %v", err)
+	}
+	s := server()
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			logrus.Fatalf("failed to serve: %v", err)
+		}
+	}()
```

Comment:
Should we call `s.GracefulStop()` at some point after the server is finished for testing purposes? Per https://godoc.org/google.golang.org/grpc#Server.Serve it will always return a non-nil error if the server is not stopped.

Reply:
It's probably good practice to cleanly stop, but this patch only aims to fix the flake that happens because the tests begin before the fixtures are ready to use.

The best path might be to find alternative ways to test this (like unit tests for each endpoint on the server) without using real gRPC clients/servers, and adding end-to-end tests that cover transport issues directly against registry-server/"opm registry serve".